### DISCUSSION TITLE
Basic SMTP Setup for Postmark

### DISCUSSION THREAD

**Discussion by benorudolf** [nhost/nhost#1532] [Tips and Tricks]
This is a quick basic setup guide for using Postmark as your email sending service when using Nhost. 

⚠️ This is not a complete guide. It doesn't cover secure connection or using other authentication methods.

### Postmark
In your Postmark account select the stream you want to use. Most commonly this is the **Default Transactional Stream**. Once in the stream secion click the **Settings** tab and scroll down to the **SMTP** area.  
 
Click on the **Generate an SMTP token** button. This will generate an **Access Token** value and a **Secret Key**. Be careful here because you will only see the secret once. Make a safe copy somewhere (don't worry, if you lose it you just need to generate a new Token). 

![CleanShot 2023-01-18 at 20 30 31@2x](https://user-images.githubusercontent.com/955836/213277046-8ef8ebd5-96f6-4e92-ac31-471e4409a1ab.png)

### Nhost
Go to your Nhost dashboard. Navigate to **Settings** and select **SMTP** from the left submenu.   
Fill in the settings as seen here: 
![CleanShot 2023-01-18 at 20 41 09@2x](https://user-images.githubusercontent.com/955836/213278881-25916f55-1944-4248-ba63-caf47a423f03.png)

From email: `<your verified Postmark sender email>`  
SMPT Host: `smtp.postmarkapp.com`    
Port: `2525`  
SMPT Username: `<the access token you generated>`
SMPT Password: `<the shy secret key you got to see once>` 
SMPT Auth method: `CRAM-MD5`  
Use SSL: Leave unchecked

👉 Generally it's not good advice to not use a secure connection if available. Some users mentioned on Discord that having this checked wasn't working for them. Your results may vary.
👉 For other available Postmark values see the [Postmark User Guide on SMTP](https://postmarkapp.com/developer/user-guide/send-email-with-smtp). The guide mentions `TLS` but using it as an Auth method value will not work.

### Check / Test
Once you save the settings it can take a couple of minutes for the changes to propagate. Go make some ☕ before testing. 

If you come back to the Nhost SMTP Settings you might see the password field empty. That's ok, the password is saved just not exposed. If you do make changes to the Settings you will need to re-enter it.

**Comment by dbarrosop** [comment]
Quick note on the `Use SSL` option. If the value is unchecked but the server supports STARTTLS the communication will still use SSL. From the docs of the underlying library we use for this (https://nodemailer.com/smtp/):


> secure – if true the connection will use TLS when connecting to server. If false (the default) then TLS is used if server supports the STARTTLS extension. In most cases set this value to true if you are connecting to port 465. For port 587 or 25 keep it false

As postmark supports STARTTLS this means the connection should automatically upgrade to use TLS.
